content/200-concepts/100-components/02-prisma-client/260-client-extensions/140-shared-extensions.mdx

```diff
@@ -7,47 +7,63 @@ tocDepth: 4
 
 </TopBlock>
 
 ## Share Prisma Client extensions
 
-When you want to create extensions that will work for other users, and not just for your specific schema, use the `$all` prefix, such as [`$allModels`](/concepts/components/prisma-client/client-extensions/model#add-a-custom-method-to-all-models-in-your-schema). Extensions that use `$all` apply generally, not just to a named component of a specific schema.
+When you want to create extensions other users can use, and not tailored just for your schema, Prisma provides utilities to allow you to create shareable extensions.
 
-You do not need to use the `$all` prefix with the [`client`](/concepts/components/prisma-client/client-extensions/client) component, because the `client` component always applies generally.
+### Define an extension
 
-For example, a generic extension might take the following form:
+Use the `Prisma.defineExtension` method to make your extension shareable. You can use it to package the extension to either separate your extensions in multiple files or share it with other users as an npm package.
+
+The benefit of `Prisma.defineExtension` is that it provides strict type checks and auto completion when you define or use an extension.
+
+To define an extension, you can use the following code snippets:
 
 ```ts
-const xprisma = prisma.$extends({
-  model: {
-    $allModels: {
-      // new method
-      findOrCreate(...) { } // code for the new method goes inside the brackets
-    }
-  }
+import { Prisma } from '@prisma/client/extension'
+
+export default Prisma.defineExtension({
+  name: 'prisma-extension-<extension-name>',
 })
 ```
 
-### Package an extension
+For versions earlier than 4.16.0, the `Prisma` import is available from a different path shown in the snippet below:
 
-Use `Prisma.defineExtension` to make your extensions shareable. You can use it to package a feature that you want to share with other users. When someone wants to use your extension in their project, they can apply the extension to a Prisma Client instance with `prisma.$extends`.
+```ts
+import { Prisma } from '@prisma/client/scripts/default-index'
 
-You can also use `Prisma.defineExtension` to move an extension to another file in your project. When you do this, you benefit from auto-completion and type checks.
+export default Prisma.defineExtension({
+  name: 'prisma-extension-<extension-name>',
+})
+```
 
-To package an extension, you might use the following code:
+### Create a generic extension method
 
-```ts
-import { Prisma } from '@prisma/client'
+To define an extension method that would work for other users, Prisma provides methods that begin with the `$all` prefix such as [`$allModels`](/concepts/components/prisma-client/client-extensions/model#add-a-custom-method-to-all-models-in-your-schema) and `$allOperations`.
+
+Extensions that use the `$all` method prefix apply generally and not to a named component of a specific schema.
```

Comment:
Weird grammar here

Reply:
I've updated the text. Let me know if it reads better.